CHANGELOG.md

```diff
@@ -26,12 +30,22 @@ of which the DTW algorithm can be run
 * `ShapeletModel` is now serializable to JSON and pickle formats
 * Multivariate datasets from the UCR/UEA archive are now available through
 `UCR_UEA_datasets().load_dataset(...)`
+* `ShapeletModel` now accepts variable-length time series dataset; a `max_size`
+parameter has been introduced to save room at fit time for possibly longer
+series to be fed to the model afterwards
+* `ShapeletModel` now accepts a `scale` parameter that drives time series
+pre-processing for better convergence
+* `ShapeletModel` now has a public `history_` attribute that stores
+loss and accuracy along fit epochs
+* SAX and variants now accept a `scale` parameter that drives time series
```

Comment:
Perhaps we could make a small note here that these will be applied on the entire dataset as opposed to the regular per-timeseries normalization?

Reply:
At the moment, it is done per time series. We should decide which one to pick and document it clearly before release of v0.4, I agree!

Reply:
If it is done per timeseries, this will only capture "shape information" because every timeseries will range from 0 to 1. Originally, it was applied on the entire dataset. Perhaps we could make this configurable?

Reply:
The dataset normalization is a little bit harder. It will require you to store a mu and sigma per dimension during fit time and re-use these during predict time to normalize.

Reply:
I agree the dataset-level normalization is the right one for SAX (and it's not very difficult to implement I think). I'll fix this is another PR.